### PR TITLE
exporter/datadogexporter: update k8s-chart example and add validation test

### PR DESCRIPTION
**Pull request by gbbr**
This change updates the current chart examples to contain a more accurate setup that functions better with Datadog. Additionally, it adds a test which ensures that the yaml can be validated with the current dependencies, ensuring that it continues to work throughout updates.

### CODE REVIEW

File: exporter/datadogexporter/examples/k8s-chart/configmap.yaml [hunk 1]
```diff
@@ -12,6 +12,22 @@ data:
...
+        # # processed, running and sent, queue sizes, uptime, k8s information
+        # # and much more.
+        #
+        # prometheus:
+        #   config:
+        #     scrape_configs:
+        #     - job_name: 'otelcol'
+        #       scrape_interval: 10s
+        #       static_configs:
+        #       - targets: ['0.0.0.0:8888']
```
Comment (gbbr) [inline exporter/datadogexporter/examples/k8s-chart/configmap.yaml:26]: @dineshg13 what is this?

File: exporter/datadogexporter/examples/k8s-chart/configmap.yaml [hunk 2]
```diff
@@ -12,6 +12,22 @@ data:
...
+        #       metric_relabel_configs:
+        #       - source_labels: [ __name__ ]
+        #         regex: '.*grpc_io.*'
```
Comment (gbbr) [inline exporter/datadogexporter/examples/k8s-chart/configmap.yaml:29]: @dineshg13 why are we ignoring this one? Id like to leave a note (or simply allow it by removing this block).

**Comment by gbbr** [comment]
Not exactly sure this warrants a changelog? @dashpole may I bug you with addressing that? Does it need a special label to indicate this? I don't have permissions to add.

File: exporter/datadogexporter/examples/k8s-chart/daemonset.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ spec:
         command:
           - "/otelcol-contrib"
           - "--config=/conf/otel-agent-config.yaml"
-        image: otel/opentelemetry-collector-contrib:0.54.0
+        image: otel/opentelemetry-collector-contrib:0.55.0
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/k8s-chart/daemonset.yaml:25]: This should at least be 0.57.0 to incorporate the breaking changes we made to the configuration and elsewhere

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-chart/daemonset.yaml]: Updated to 0.59.0

File: exporter/datadogexporter/examples/collector.yaml [hunk 4]
```diff
@@ -13,6 +13,55 @@ receivers:
       # The presence of the gRPC section enables the gRPC server on the default port (4317)
       grpc:
 
+  # k8s_cluster receiver scrapes valuable Kubernetes metrics.
+  k8s_cluster:
+    collection_interval: 10s
```
Comment (dineshg13) [inline exporter/datadogexporter/examples/collector.yaml:18]: This needs appropriate roles to be setup https://github.com/DataDog/opentelemetry-helm-charts/blob/main/charts/opentelemetry-demo-datadog/templates/roles.yaml

**Review by dineshg13 — CHANGES_REQUESTED** [review]

File: exporter/datadogexporter/examples/k8s-chart/daemonset.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-chart/daemonset.yaml]: It's the latest. Updated 20 min. ago. You might have had a stale tab open 🙁

File: exporter/datadogexporter/examples/k8s-chart/configmap.yaml [hunk 5]
```diff
@@ -11,7 +11,50 @@ data:
...
+            metrics:
+              system.memory.utilization:
+                enabled: true
+          network:
+            metrics:
+              system.network.io.receive:
+                enabled: true
+              system.network.io.transmit:
+                enabled: true
+          processes:       http:
```
Comment (mx-psi) [inline exporter/datadogexporter/examples/k8s-chart/configmap.yaml:44]: This line looks off, is there a newline missing maybe?

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-chart/configmap.yaml]: Damn. Fixed.

**Review by mx-psi — APPROVED** [review]
This LGTM, but I think we need some validation so that this is kept up to date

File: exporter/datadogexporter/go.mod [hunk 6]
```diff
@@ -15,13 +15,17 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig v0.59.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders v0.59.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry v0.59.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor v0.59.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor v0.59.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver v0.59.0
```
Comment (mx-psi) [inline exporter/datadogexporter/go.mod:20]: You need to add a replace statement for each of these so that the latest changes are picked up

Comment (gbbr) [reply exporter/datadogexporter/go.mod:20]: Yes. I have to get into the habit of this. Done.

**Review by dineshg13 — APPROVED** [review]
LGTM

**Review by mx-psi — APPROVED** [review]

**Pull request merged**